test/protocol/modules/CustomOracleNAVIssuanceModule.spec.ts

```diff
@@ -0,0 +1,3541 @@
+        setToken = await setup.createSetToken(
+          [setup.weth.address, setup.usdc.address, setup.wbtc.address, setup.dai.address],
+          units, // Set is valued at 2000 USDC
+          [setup.issuanceModule.address, customOracleNavIssuanceModule.address]
+        );
+        const managerIssuanceHook = await getRandomAddress();
+        const reserveAssets = [setup.usdc.address, setup.weth.address];
+        const managerFeeRecipient = feeRecipient.address;
+        // Set max managerFee to 20%
+        const maxManagerFee = ether(0.2);
+        // Set max premium to 10%
+        const maxPremiumPercentage = ether(0.1);
+        // Set min SetToken supply required
+        const minSetTokenSupply = ether(1);
+
+        navIssuanceSettings = {
+          managerIssuanceHook,
+          managerRedemptionHook,
+          reserveAssets,
+          setValuer: setValuerAddress,
+          feeRecipient: managerFeeRecipient,
+          managerFees,
+          maxManagerFee,
+          premiumPercentage,
+          maxPremiumPercentage,
+          minSetTokenSupply,
+        } as CustomOracleNAVIssuanceSettings;
+
+        await customOracleNavIssuanceModule.initialize(setToken.address, navIssuanceSettings);
+        // Approve tokens to the controller
+        await setup.weth.approve(setup.controller.address, ether(100));
+        await setup.usdc.approve(setup.controller.address, usdc(1000000));
+        await setup.wbtc.approve(setup.controller.address, bitcoin(1000000));
+        await setup.dai.approve(setup.controller.address, ether(1000000));
+
+        // Seed with 10 supply
+        await setup.issuanceModule.connect(owner.wallet).initialize(setToken.address, ADDRESS_ZERO);
+        await setup.issuanceModule.connect(owner.wallet).issue(setToken.address, ether(10), owner.address);
+
+        // Redeem 1 SetToken
+        redeemQuantity = ether(2.8);
+
+        subjectSetToken = setToken.address;
+        subjectReserveAsset = setup.usdc.address;
+        subjectSetTokenQuantity = redeemQuantity;
+        subjectMinReserveQuantityReceived = ether(0);
+        subjectTo = recipient;
+        subjectCaller = owner;
+      });
+
+      context("when using a custom set valuer", () => {
+        before(async () => {
+          managerRedemptionHook = ADDRESS_ZERO;
+          // Set fees to 0
+          managerFees = [ether(0), ether(0)];
+          // Set premium percentage to 50 bps
+          premiumPercentage = ether(0);
+          const setValuerMock = await deployer.mocks.deployCustomSetValuerMock();
+          // set valued at $500 by the custom set valuer
+          // which values ETH at 200 usd and usdc at 1 usd
+          await setValuerMock.setValuation(setup.usdc.address, ether(370));
+          await setValuerMock.setValuation(setup.weth.address, ether(1.85)); // 370/200
+          setValuerAddress = setValuerMock.address;
+        });
+
+        const  subject = () =>
```

Comment:
Same here as the comments on `#issue` let's use either new `subjectVariables` or can use `getExpectedReserveRedeemQuantity` found on line 2318.